crates/red_knot_python_semantic/src/semantic_index/expression.rs

```diff
@@ -51,6 +51,17 @@ pub(crate) struct Expression<'db> {
 }
 
 impl<'db> Expression<'db> {
+    /// Returns a reference to the expression's AST node.
+    ///
+    /// `query_file` is the file for which the current query performs type inference.
+    /// It acts as a token of prove that we aren't accessing an AST node from a different file
+    /// than in which the current enclosing Salsa query (which would lead to cross-file dependencies).
```

Comment:
```suggestion
    /// It acts as a token of proof that we aren't accessing an AST node from a different file
    /// to the one from which the current enclosing Salsa query is being called.
    /// Doing so would lead to cross-file dependencies, hurting incremental computation.
```